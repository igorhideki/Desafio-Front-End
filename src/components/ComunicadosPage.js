import React, { Component } from 'react';
import CategoriaFilters from './CategoriaFilters';
import ComunicadoList from './ComunicadoList';
import api from '../requests/apis';
import selectComunicados from '../selectors/comunicados';

class ComunicadosPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      categorias: [],
      comunicados: [],
      filters: {
        type: -1
      }
    };
    this.handleSelectCategoria = this.handleSelectCategoria.bind(this);
  }

  componentDidMount () {
    this.setState({ isLoading: true });
    api.loadCategorias()
      .then((res) => {
        this.setState(() => ({ isLoading: false, categorias: res.data }));
      });
    api.loadComunicados()
      .then((res) => {
        this.setState(() => ({ isLoading: false, comunicados: res.data }));
      });
  }

  handleSelectCategoria(type) {
    if (this.state.filters.type === type) {
      this.setState(() => ({ filters: { type: -1 } }));
    } else {
      this.setState(() => ({ filters: { type } }));
    }
  }

  render() {
    return (
      <div className="container">
        <CategoriaFilters categorias={this.state.categorias} handleSelectCategoria={this.handleSelectCategoria} filterActive={this.state.filters.type}/>
        <ComunicadoList comunicados={selectComunicados(this.state.comunicados, this.state.filters)}/>
      </div>
    );
  }
} 

export default ComunicadosPage;
