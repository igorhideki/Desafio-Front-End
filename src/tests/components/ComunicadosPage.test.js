import React from 'react';
import { shallow } from 'enzyme';
import ComunicadosPage from '../../components/ComunicadosPage';

test('should render ComunicadosPage correctly', () => {
  const wrapper = shallow(<ComunicadosPage />);
  expect(wrapper).toMatchSnapshot();
});
