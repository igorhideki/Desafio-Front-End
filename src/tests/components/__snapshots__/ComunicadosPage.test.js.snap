// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadosPage correctly 1`] = `
<div
  className="container"
>
  <span>
    <i
      className="fas fa-spinner fa-pulse mr-xs"
    />
     Aguarde, carregando...
  </span>
</div>
`;
